### PR TITLE
VCB-151 Fix store instruction

### PR DESCRIPTION
**Pull request by m-dot-wigger**
can be merged after #33 

example code
```
	AREA toCheck, DATA, READWRITE
myVar        		DCD     0x12345678
AREA abc, CODE, READONLY
b
				
      LDR R1, =0xffffffff
      LDR R0, =0x0
      LDR R2, =myVar
      LDR R2, =myVar
      
      STR R0, [R2]
      STRH R1, [R2]
      STRB R0, [R2]
      B b

```

### CODE REVIEW

**Comment by ni507** [comment]
@m-dot-wigger please merge the master branch into this branch and make the PR review "ready" after #33 is merged.

**Comment by m-dot-wigger** [comment]
@ni507 is at the moment review ready until next PR is merged into master

File: src/instruction/instructions/store/str.ts [hunk 1]
```diff
@@ -25,19 +25,35 @@ export class StrImmediate5OffsetInstruction extends BaseInstruction {
...
+    new StrRegisterOffsetInstruction()
+  ]
 
   public canEncodeInstruction(name: string, options: string[]): boolean {
     return (
       super.canEncodeInstruction(name, options) &&
-      isOptionCountValid(options, this.expectedOptionCount) &&
-      isImmediate(options[2]) &&
-      registerStringHasBrackets(options[1], options[2])
+      !this.otherInstructionWithSameName.some((instr) => {
```
Comment (ni507) [inline src/instruction/instructions/store/str.ts:37]: IMO can be on one line. `!this.otherInstructionWithSameName.some((instr) => instr.canEncodeInstruction(name, options))`

File: src/instruction/instructions/store/str.ts [hunk 2]
```diff
@@ -25,19 +25,35 @@ export class StrImmediate5OffsetInstruction extends BaseInstruction {
...
+      options,
+      this.expectedOptionCountMin,
+      this.expectedOptionCountMax
+    )
+    checkBracketsOnLastOptions(
+      options,
+      this.expectedOptionCountMin,
+      this.expectedOptionCountMax
+    )
+    if (options.length == this.expectedOptionCountMin) {
```
Comment (ni507) [inline src/instruction/instructions/store/str.ts:54]: use `===` (strict equals operator) for comparison

**Review by ni507 — CHANGES_REQUESTED** [review]
Changes must be applied for all instructions (`STR`, `STRB`, ``STRH`)

File: src/instruction/instructions/store/str.ts (hunk 1 above)
Comment (m-dot-wigger) [reply src/instruction/instructions/store/str.ts]: I had to tweak a little bit to make it a one liner

File: src/instruction/instructions/store/str.ts (hunk 2 above)
Comment (m-dot-wigger) [reply src/instruction/instructions/store/str.ts]: done also for load instructions and created issue for remaining code https://virtualctboard.atlassian.net/browse/VCB-215

**Review by ni507 — APPROVED** [review]

**Pull request merged**